### PR TITLE
Add parser for multiple choice questions

### PR DESCRIPTION
**Pull request by radwouters**
Add parser for questions with multiple answer possibilities, such as a checkbox list

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -35,9 +35,12 @@ jobs:
     name: commit-lint
     runs-on: ubuntu-latest
     steps:
-      - uses: ahmadnassri/action-commit-lint@v1
+      - uses: actions/checkout@v3
         with:
-         config: conventional
+          fetch-depth: 0
```
Comment (SamMousa) [inline .github/workflows/ci.yaml:40]: If you use fetch depth 0 how is the commitlint action ever going to check older commits?

File: src/Parsers/MultipleChoiceQuestionParser.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            } elseif (is_scalar($value)) {
+                $result[] = new StringValueOption((string) $value, $displayValues);
+            } else {
+                throw new \InvalidArgumentException('Values must be scalar, got: ' . print_r($choice, true));
+            }
+        }
+
+        return $result;
+    }
+}
```
Comment (SamMousa) [inline src/Parsers/MultipleChoiceQuestionParser.php:75]: Isn't this already defined in the `ParserHelpers` trait?

File: src/Variables/SingleChoiceVariable.php [hunk 3]
```diff
@@ -26,7 +26,7 @@ class SingleChoiceVariable implements ClosedVariableInterface
     /**
      * @param string $name
      * @param array<string, string> $titles
-     * @phpstan-param non-empty-list<ValueOptionInterface> $valueOptions
+     * @phpstan-param array<ValueOptionInterface> $valueOptions
```
Comment (SamMousa) [inline src/Variables/SingleChoiceVariable.php:29]: This is not a keyed array; so if it really can not be `non-empty-list` then it should be `list`. Why was the old annotation wrong?

File: tests/Parsers/MultipleChoiceQuestionParserTest.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                ['value' => 'a', 'text' => 'b'],
+                'c',
+                15,
+                ['value' => 16, 'text' => 'abc']
+            ],
+            'name' => 'q1',
+        ];
+
+        $variable = toArray($parser->parse($parent, $questionConfig, $surveyConfiguration));
+        self::assertInstanceOf(MultipleChoiceVariable::class, $variable[0]);
```
Comment (SamMousa) [inline tests/Parsers/MultipleChoiceQuestionParserTest.php:44]: This seems like a lazy and false way to get coverage.
You've confirmed that the parser doesn't crash, but you haven't tested anything related to the config you passed.

**Review by SamMousa — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (radwouters) [reply .github/workflows/ci.yaml:40]: See [readme of the checkout action](https://github.com/actions/checkout): 0 means all commits

File: tests/Parsers/MultipleChoiceQuestionParserTest.php (hunk 4 above)
Comment (radwouters) [reply tests/Parsers/MultipleChoiceQuestionParserTest.php]: Yes, good point.

File: tests/Values/ValueSetTest.php [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+    public function testGetValueSetValues(): void
+    {
+        $values = [
+           new StringValueOption('test', ['nl' => 'test']),
+           new StringValueOption('test2', ['nl' => 'test2']),
+        ];
+        $set = new ValueSet(...$values);
+
+        self:assertEquals($values, $set->getValues());
```
Comment (SamMousa) [inline tests/Values/ValueSetTest.php:45]: This is wrong

File: tests/Values/ValueSetTest.php [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+declare(strict_types=1);
+
+namespace Collecthor\SurveyjsParser\Tests\Variables;
+
+use Collecthor\SurveyjsParser\Values\StringValue;
+use Collecthor\SurveyjsParser\Values\StringValueOption;
+use Collecthor\SurveyjsParser\Values\ValueSet;
+use PHPUnit\Framework\TestCase;
+
+use function PHPUnit\Framework\assertEquals;
```
Comment (SamMousa) [inline tests/Values/ValueSetTest.php:12]: ?

File: tests/Variables/MultipleChoiceVariableTest.php [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+ * @covers \Collecthor\SurveyjsParser\Variables\MultipleChoiceVariable
+ * @uses \Collecthor\SurveyjsParser\Values\IntegerValueOption
+ * @uses \Collecthor\SurveyjsParser\Values\StringValueOption
+ * @uses \Collecthor\SurveyjsParser\ArrayRecord
+ * @uses \Collecthor\SurveyjsParser\Values\StringValue
+ * @uses \Collecthor\SurveyjsParser\Values\InvalidValue
+ * @uses \Collecthor\SurveyjsParser\ArrayDataRecord
+ * @uses \Collecthor\SurveyjsParser\Values\ValueSet
+ */
+class MultipleChoiceVariableTest extends TestCase
```
Comment (SamMousa) [inline tests/Variables/MultipleChoiceVariableTest.php:28]: final

**Review by SamMousa — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml [hunk 8]
```diff
@@ -37,3 +37,10 @@ jobs:
       - uses: wagoid/commitlint-github-action@v5
         with:
           configFile: ../../commitlint.config.js
+  release:
+    name: Automated release
+    needs: [code_checks]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
```
Comment (SamMousa) [inline .github/workflows/ci.yaml:46]:
```suggestion
```

**Review by SamMousa — APPROVED** [review]

**Pull request merged**